src/main/java/hudson/plugins/analysis/core/AbstractResultAction.java

```diff
@@ -104,15 +111,29 @@ public ToolTipProvider getToolTipProvider() {
      *
      * @return the associated build of this action
      */
-    public final AbstractBuild<?, ?> getOwner() {
+    @WithBridgeMethods(value=AbstractBuild.class, adapterMethod="getAbstractBuild")
+    public final Run<?, ?> getOwner() {
         return owner;
     }
 
     @Override
-    public final AbstractBuild<?, ?> getBuild() {
+    @WithBridgeMethods(value=AbstractBuild.class, adapterMethod="getAbstractBuild")
+    public final Run<?, ?> getBuild() {
         return owner;
     }
 
+    /**
+     * Added for backward compatibility. It generates <pre>AbstractBuild getOwner()</pre> bytecode during the build
+     * process, so old implementations can use that signature.
+     * 
+     * @see {@link WithBridgeMethods}
+     */
+    @Restricted(NoExternalUse.class)
+    @Deprecated
+    public final Object getAbstractBuild(Run owner, Class targetClass) {
+      return owner instanceof AbstractBuild ? (AbstractBuild) owner : null;
+    }
```

Comment:
When this method is used by old code shouldn't there be rather a CCE then a NPE? Would help debugging.